src/app/login.service.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Injectable} from '@angular/core';
+import {User} from './user';
+import {environment} from './../environments/environment';
+import {HttpStalkerService} from './http-stalker.service';
+import {catchError} from 'rxjs/operators';
+import {Observable, of} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoginService {
+  // loginURL = environment.apiUrl + 'user/login';
+  loginURL = environment.apiUrl + '/user/login';
+  constructor(private httpStalker: HttpStalkerService) {}
+  login(user: User): any {
+    return this.httpStalker
+      .fakepost(this.loginURL, user)
+      .pipe(catchError(this.handleError<any>([])));
+  }
```

Comment:
questa sintassi qua per i servizi è artificialmente complicata: le cose fatte "a regola d'arte" dovrebbero essere pesantemente asincrone e utilizzare il publisher/subscriber, 

```ts
getHeroes(): void {
  this.heroService.getHeroes()
      .subscribe(heroes => this.heroes = heroes);
}
```

dal tutorial https://angular.io/tutorial/toh-pt4#subscribe-in-heroescomponent

Reply:
da noi infatti il subscribe avviene nel component, il service si limita a preparare la chiamata al servizio che estende httpClient e restituire il risultato a component che puoi farà subscribe del risultato

Reply:
mmm forse allora non ho capito io un paio di cose

Reply:
questo commento è nel file del loginService, il loginService chiama la funzione di httpStalkerService che gli restituisce un'Observable, il loginService lo prende così com'è e lo passa al chiamante, in questo caso loginComponent che fa il subscribe